src/main/scala/mesosphere/marathon/core/task/tracker/impl/steps/ContinueOnErrorStep.scala

```diff
@@ -18,13 +18,21 @@ class ContinueOnErrorStep(wrapped: TaskStatusUpdateStep) extends TaskStatusUpdat
   override def name: String = s"continueOnError(${wrapped.name})"
 
   override def processUpdate(
-    timestamp: Timestamp, appId: PathId, maybeTask: Option[MarathonTask], mesosStatus: TaskStatus): Future[_] = {
+    timestamp: Timestamp, appId: PathId, task: MarathonTask, mesosStatus: TaskStatus): Future[_] = {
 
     import scala.concurrent.ExecutionContext.Implicits.global
-    wrapped.processUpdate(timestamp, appId, maybeTask, mesosStatus).recover {
-      case NonFatal(e) =>
-        log.error("while executing step {} for [{}], continue with other steps",
-          wrapped.name, mesosStatus.getTaskId.getValue, e)
+    val maybeProcessed: Option[Future[_]] = Option(wrapped.processUpdate(timestamp, appId, task, mesosStatus))
+    maybeProcessed match {
+      case Some(processed) =>
+        processed.recover {
+          case NonFatal(e) =>
+            log.error("while executing step {} for [{}], continue with other steps",
+              wrapped.name, mesosStatus.getTaskId.getValue, e)
+        }
+      case None =>
```

Comment:
We are using a test framework that leads to returning "null" values when a call is not mocked. When the code changes and the test fails, not handling this case explicitly led to a situation that was hard to debug. This code makes it easy.

In addition, the code does not make the production code less robust -- it actually gets more robust.

If you really insist, we'll remove the code but we both would prefer to keep it.